src/vs/workbench/api/browser/mainThreadLocalization.ts

```diff
@@ -0,0 +1,27 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MainContext, MainThreadLocalizationShape } from 'vs/workbench/api/common/extHost.protocol';
+import { extHostNamedCustomer, IExtHostContext } from 'vs/workbench/services/extensions/common/extHostCustomers';
+import { URI, UriComponents } from 'vs/base/common/uri';
+import { IFileService } from 'vs/platform/files/common/files';
+import { retry } from 'vs/base/common/async';
+import { Disposable } from 'vs/base/common/lifecycle';
+
+@extHostNamedCustomer(MainContext.MainThreadLocalization)
+export class MainThreadLocalization extends Disposable implements MainThreadLocalizationShape {
+
+	constructor(
+		extHostContext: IExtHostContext,
+		@IFileService private readonly fileService: IFileService,
+	) {
+		super();
+	}
+
+	async $fetchBundleContents(uriComponents: UriComponents): Promise<string> {
+		const contents = await retry(() => this.fileService.readFile(URI.revive(uriComponents)), 50, 3);
```

Comment:
😕 why do you need `retry` here? Did you encounter a case where the file service doesn't work at first try?

Reply:
It's not needed. I added it for resiliency just in case, but I can remove it